src/gr11review/part2/Utility.java

```diff
@@ -1,8 +1,348 @@
+     * @param str, the string that is being tested in the method
+     * @returns a boolean of true or false 
+     * @author K. Sinclair
+     */
+
+    public static boolean xyzMiddle(String str){
+        // define the return boolean
+        boolean isXYZMiddle;
+    
+        // when the string contains 'xyz' and length of string is odd
+        if (str.contains("xyz") && (str.length() % 2 != 0)){
+
+            // create character placement expectations for 'xyz' in the middle
+            int intYPlacement = (str.length() - 1) / 2;
+            int intXPlacement = intYPlacement - 1;
+            int intZPlacement = intYPlacement + 1;
+
+            // check if the character placements are in the middle
+            if ((str.charAt(intYPlacement) == 'y') && (str.charAt(intXPlacement) == 'x') && (str.charAt(intZPlacement) == 'z')){
+                isXYZMiddle = true;
+            }
+            else{
+                isXYZMiddle = false;
+            }
+        }
+        else{
+            isXYZMiddle = false;
+        }
+        return isXYZMiddle;
+    }
+    
+    /**
+     * Given a file with one word on each line, return the word with the most characters
+     * 
+     * @param filenametxt The name of the file
+     * @return strLongestWord, the word with the most characters in the text file
+     * @author D. Gu
+     */
+    public static String longestWord(String filenametxt) throws IOException {
+        // Initialize characters in the longest word to be 0
+        int intCharacters = 0;
+        String strCompare = "";
+        String strLongestString = "";
+        Scanner myScanner = new Scanner(new File(filenametxt));
+        
+        // As long as the txt file has another line, set it equal to strCompare
+        while (myScanner.hasNextLine()) {
+            strCompare = myScanner.nextLine();
+
+            // If there are more characters in strCompare, it is the new longest word
+            if (strCompare.length() > intCharacters) {
+                strLongestString = strCompare;
+                intCharacters = strLongestString.length();
+            }
+        }
+        // Close the scanner and return the longest word
+        myScanner.close();
+        return strLongestString;
+    }
+
+    /**
+     * File IO - Read 2
+     * Evaluates a list of words and determines which word will come first in alphabetical order
+     * 
+     * @param filenametxt, the file that contains the list of names being tested
+     * @returns strAlphaWord, the word that comes first alphabetically
+     * @author K. Sinclair
+     */
+    
+    public static String alphaWord (String filenametxt) throws IOException{
+        // read the file
+        Scanner scanFile = new Scanner(new File(filenametxt));
+        
+        // define variables
+        String strWord = "";
+        String strAlphaWord = "z";
+
+        // go through all the words in the file 
+        while (strWord != null) {
+            strWord = scanFile.nextLine();
 
```

Comment:
your program runs into trouble here when it reaches the end of the file and tries to process a null value.  put your file reading commands at the end of the loop